R/analysis.R

```diff
@@ -1,63 +1,96 @@
     ) %>%
-    return()
+    # Replace Inf or NaN with NA and warn
+    dplyr::mutate(
+      yi_adj = dplyr::if_else(is.finite(yi_adj), yi_adj, NA_real_),
+      vi_adj = dplyr::if_else(is.finite(vi_adj), vi_adj, NA_real_)
+    )
 
+  return(dat)
 }
 
-#' Title
+#' Prepare bias and indirectness data for adjustment
+#'
+#' Aggregates domain-level prior data and calculates total adjustment values
+#' per study result. Supports additive and proportional adjustments.
 #'
-#' @param dat_bias Cleaned bias dataset
-#' @param dat_ind Cleaned indirectness dataset
+#' @param dat_bias Data with bias priors (long format)
+#' @param dat_ind Data with indirectness priors (long format)
+#'
+#' @return Data with yi/vi and adjustment components ready for adjustment
 #'
 #' @export
-tri_prep_data <- function(dat_bias, dat_ind){
+tri_prep_data <- function(dat_bias, dat_ind) {
 
   i_add <- dat_bias %>%
     dplyr::group_by(result_id) %>%
-    dplyr::summarise(addimn = sum(bias_m_add, na.rm = T),
-              addivar = sum(bias_v_add, na.rm = T)) %>%
-    dplyr::select(result_id, dplyr::starts_with("add"))
+    dplyr::summarise(
+      addimn = sum(bias_m_add, na.rm = TRUE),
+      addivar = sum(bias_v_add, na.rm = TRUE),
+      .groups = "drop"
+    )
 
   e_add <- dat_ind %>%
     dplyr::group_by(result_id) %>%
-    dplyr::summarise(addemn = sum(ind_m_add, na.rm = T),
-              addevar = sum(ind_v_add, na.rm = T)) %>%
-    dplyr::select(result_id, dplyr::starts_with("add"))
+    dplyr::summarise(
+      addemn = sum(ind_m_add, na.rm = TRUE),
+      addevar = sum(ind_v_add, na.rm = TRUE),
+      .groups = "drop"
+    )
 
   i_prop <- dat_bias %>%
     dplyr::group_by(result_id) %>%
-    dplyr::summarise(sumlogmn = sum(bias_m_prop, na.rm = T),
-              sumlogvr = sum(bias_v_prop, na.rm = T),
-              propimn = exp(sumlogmn+sumlogvr/2),
-              propivar = (exp(2*sumlogmn+sumlogvr)*(exp(sumlogvr)-1))) %>%
-    dplyr::select(result_id, dplyr::starts_with("prop"))
+    dplyr::summarise(
+      sumlogmn = sum(bias_m_prop, na.rm = TRUE),
+      sumlogvr = sum(bias_v_prop, na.rm = TRUE),
+      propimn = exp(sumlogmn + sumlogvr / 2),
+      propivar = exp(2 * sumlogmn + sumlogvr) * (exp(sumlogvr) - 1),
+      .groups = "drop"
+    )
 
   e_prop <- dat_ind %>%
     dplyr::group_by(result_id) %>%
-    dplyr::summarise(sumlogmn = sum(ind_m_prop, na.rm = T),
-              sumlogvr = sum(ind_v_prop, na.rm = T),
-              propemn = exp(sumlogmn+sumlogvr/2),
-              propevar = (exp(2*sumlogmn+sumlogvr)*(exp(sumlogvr)-1))) %>%
-    dplyr::select(result_id, dplyr::starts_with("prop"))
+    dplyr::summarise(
+      sumlogmn = sum(ind_m_prop, na.rm = TRUE),
+      sumlogvr = sum(ind_v_prop, na.rm = TRUE),
+      propemn = exp(sumlogmn + sumlogvr / 2),
+      propevar = exp(2 * sumlogmn + sumlogvr) * (exp(sumlogvr) - 1),
+      .groups = "drop"
+    )
 
   dat_final <- dat_bias %>%
     dplyr::distinct(result_id, study, yi, vi) %>%
```

Comment:
It would be useful if `tri_prep_data()` also returned the `type` variable, to allow for down-stream stratification by study type (if needed)